.goreleaser.yml

```diff
@@ -0,0 +1,52 @@
+project_name: driftctl
+release:
+    github:
+        owner: cloudskiff
+        name: driftctl
+    draft: false
+    prerelease: auto
+before:
+    hooks:
+        - go mod tidy
+        - go generate ./...
+signs:
+    - signature: "${artifact}.gpg"
+      artifacts: checksum
+      args: [ "--batch", "-u", "{{ .Env.SIGNINGKEY }}", "--output", "${signature}", "--detach-sign", "${artifact}" ]
+builds:
+    - id: "driftctl-build"
+      binary: driftctl
+      dir: .
+      env:
+          - CGO_ENABLED=0
+          - GO111MODULE=on
+      goos:
+          - linux
+          - windows
+          - darwin
+      goarch:
+          - amd64
+          - arm
+          - arm64
+          - 386
+      ldflags: "-X github.com/cloudskiff/driftctl/pkg/version.version={{ .Tag }}"
```

Comment:
We should have a single entrypoint for the build to be able to reproduce the same behavior in CI and locally. Previously we have a `build.sh` script which responsibility was to pass proper flags. We should find a way to be able to build things locally too. What's the behavior of goreleaser if we are not on a tag ? Is there a way to wrap goreleaser in the `build.sh` script to get the same logic as before ? BTW it misses the flag `-X github.com/cloudskiff/driftctl/build.env=release`. I don't think we can avoid a custom build script and thus it should be updated to to not use gox anymore. We should absolutely have a single way to build binaries to avoid the `work on my machine` effect. I stop the review here since it will probably implies a lot of changes

Reply:
I agree but goreleaser can be used in local environments too. I could make a script that simply uses goreleaser to build locally. WDYT ?

> What's the behavior of goreleaser if we are not on a tag ?

I think the default value is the current commit hash (I'll try to confirm)

Reply:
https://goreleaser.com/cmd/goreleaser_build/

Reply:
> I agree but goreleaser can be used in local environments too. I could make a script that simply uses goreleaser to build locally. WDYT ?

Yeah i think it's fine if you adapt the existing script to make it work with goreleaser that's the idea. We need to have a single way to build a binary for : 
- Docker: release mode, version tag
- local:  dev mode, commit hash (if not checked out on a tag), not cross platforrm (build only the current platform, but with the availability to cross build another specific platform, for example to build a binary with a hotfix to be tested by people like we already did before on windows) 
- CI: release mode, version tag, GPG signature

Reply:
I created a script for each use case : build and release, let me know what you think about it

- Build script builds the project for the current platform (intended for devs)
- Release script builds the project for all platforms, sign using GPG, publish on github if needed (intended for CI)